src/wwwroot/app/services/EventsService.js

```diff
@@ -70,8 +70,9 @@ export default class EventsService {
 
    _convertFromServerFormat(event) {
       event.eventDate = utils.formatDateTimeFromServer(event.eventDate);
-      this._fillEntities(event);
-      this._convertIdsToString(event);
+      this._fillEntities(event).then(() => {
```

Comment:
you can call method in 'then' instead of creating a new clause. like 
**this._fillEntities(...).then(this._convertIdsToString(event));**